lib/puppet/module_tool/metadata.rb

```diff
@@ -102,8 +102,8 @@ def to_json
       data = @data.dup.merge('dependencies' => dependencies)
 
       contents = data.keys.map do |k|
-        value = (JSON.pretty_generate(data[k]) rescue data[k].to_json)
-        "#{k.to_json}: #{value}"
+        value = (Puppet::Util::Json.dump(data[k], :pretty => true) rescue Puppet::Util::Json.dump(data[k]))
```

Comment:
Is there a place where we'd raise trying to pretty print something but then can print it ugly? I assume the `data[k]` could be an object that the JSON pretty printer doesn't know how to print but has `to_json` defined on it?

Reply:
I was wondering about this too. I don't really know enough about the types involved to be sure...

Reply:
I looked through the code and it looks like we try to print an array of Dependency objects, Dependency defines to_data_hash and mixes in Puppet::Network::FormatSupport. FormatSupport defines to_json to call to_data_hash and then dump.

Reply:
I've gone through the blame on this, and the commit where the rescue was originally added (which also added all the surrounding content https://github.com/Magisus/puppet/commit/41c88a2732b546f570414ff06279b3be1393e7e2) is silent on why it was needed. Not sure how comfortable I feel taking it out though.

Reply:
Regardless of the types, I wouldn't expect any object that raises when pretty printed to not raise again printed w/o the pretty formatting. I think the important thing in this rescue is the switch from trying to use JSON's pretty printing to deferring to a method the object can implement itself.

Reply:
We still need to move this back to calling `.to_json` in the rescue, right?

Reply:
Oh sorry I misunderstood. I guess that's safe so long as we commit to always pulling in those monkey patches.

Reply:
Updated.

Reply:
I think I also agree that it seems much less likely that this rescue will help in this world. Previously you were doing two entirely separate operations, now they don't seem as different?